common/string_utils.cc

```diff
@@ -79,6 +80,9 @@ bool SplitString(const std::string &str,
   *part_2 = str.substr(pos+1);
   return true;
 }
+std::string Base64Encode(const unsigned char* data, size_t len) {
+  return g_base64_encode(data, len);
```

Comment:
returned encoded data from g_base64_encode() should be freed with g_free() or other free functions..
you can use decl_type of std::unique_ptr here.